azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/device/registration_client_factory.py

```diff
@@ -0,0 +1,37 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides factory methods for creating clients which would
+communicate with Device Provisioning Service.
+"""
+from ..security.sk_security_provider import SymmetricKeySecurityProvider
+from ..transport.mqtt_transport import MQTTTransport
+from .sk_registration_client import SymmetricKeyRegistrationClient
+
+
+def create_from_security_provider(provisioning_host, security_provider, transport_choice):
+    """
+    Creates different types of registration clients which can enable devices to communicate with Device Provisioning
+    Service based on parameters passed.
+    :param provisioning_host: Host running the Device Provisioning Service. Can be found in the Azure portal in the
+    Overview tab as the string Global device endpoint
+    :param security_provider: Instance of Security client object which can be either of Symmetric Key, TPM or X.509
```

Comment:
>security_provider [](start = 11, length = 17)

the param is called security provider but the comment talks about security client. other sdks talk about security client.. i would suggest to go with this until we decide another name is better?

Reply:
changed to security client